src/screens/registerProfessor/components/hooks/useRegister.ts

```diff
@@ -0,0 +1,106 @@
+import React, { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { SCREENS } from '../../../../utils/screens';
+import useRegisterProfessorRequest from '../../../../service/requests/useRegisterProfessorRequest';
+import { TRegisterProfessorHook } from './types';
+
+const useRegisterProfessor = (): TRegisterProfessorHook => {
+  const navigate = useNavigate();
+  const { error, isLoading, isSuccess, register, resetError } =
+    useRegisterProfessorRequest();
+
+  const [emailError, setEmailError] = useState('');
+  const [passwordError, setPasswordError] = useState('');
+  const [nameError, setNameError] = useState('');
+  const [confPasswordError, setConfPasswordError] = useState('');
+  const [password, setPassword] = useState('');
+  const [confPassword, setConfPassword] = useState('');
+  const [email, setEmail] = useState('');
+  const [name, setName] = useState('');
+
+  const handlePasswordChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setPassword(e.target.value);
+  };
+  const handleConfPasswordChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setConfPassword(e.target.value);
+  };
+  const handleEmailChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setEmail(e.target.value);
+  };
+  const handleNameChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setName(e.target.value);
+  };
+
+  const handleLoginClick = () => {
+    navigate(SCREENS.HOME);
+  };
+
+  const handleContinueClick = (e: React.SyntheticEvent<EventTarget>) => {
+    e.preventDefault();
+    setPasswordError('');
+    setEmailError('');
+    resetError();
+
+    if (!name) {
+      return setNameError('Informe o nome');
+    }
+
+    if (name.indexOf('') === -1) {
+      return setNameError('Nome inválido');
+    }
+
+    if (!email || email.indexOf('@') === -1) {
+      return setEmailError('Informe um e-mail válido');
+    }
+
+    if (email.indexOf('@icomp.ufam.edu.br') === -1) {
+      return setEmailError('Este e-mail não pertence ao domínio do icomp');
```

Comment:
IComp escrito errado:
```suggestion
      return setEmailError('Este e-mail não pertence ao domínio do IComp');
```